### PR TITLE
Service must exist Ansible 2.0

### PR DESCRIPTION
**Pull request by retr0h**
This brings a couple pull request to Ansible 2.0 which were originally accepted onto master, but later reverted.
#751
#2149
#2724

/cc @j2sol

### CODE REVIEW

File: system/service.py [hunk 1]
```diff
@@ -74,6 +74,14 @@
         description:
         - Additional arguments provided on the command line
         aliases: [ 'args' ]
+    must_exist:
+        required: false
+        default: true
+        version_added: "1.9"
```
Comment (alxgu) [inline system/service.py]: change it to 2.1

**Comment by gregdek** [comment]
Thanks @retr0h for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

**Comment by omgjlk** [comment]
:+1:

**Comment by omgjlk** [comment]
Friendly ping here to add this stuff back in.

**Comment by gregdek** [comment]
Thanks again to @retr0h for this PR, and thanks @grossws for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by grossws** [comment]
@gregdek, not guilty! I mean, I haven't reviewed this patch (it was tagged with `core_review` since long ago and was reviewed by @j2sol) and ansibull-bot message is quite misleading.

**Comment by retr0h** [comment]
@grossws Could we get eyes on this?

**Comment by grossws** [comment]
@retr0h, I'll try to take a look this week.

**Comment by retr0h** [comment]
@grossws I know you guys are busy, but how am I supposed to get features landed upstream, and not use long running local forks, when I can't get something merged in ~7 months?  I'm trying to do the right thing and get off a local fork.

**Comment by grossws** [comment]
@j2sol, @gregdek, I know that current tag is `core_review` but I review a patch, and my resolution is just `shipit`. It would be great if you can push it forward.

@retr0h, sorry for incovenience and delay. Merge timeframe of several months seems to be awful and my patches were merged in similar timeframe in past.
I'm not Ansible/RH employee but just an ansible user who accepted burden of reviewing/maintaining several ansible modules. So I can review PRs only when have some free time for it but last month was really hard time at work. If you have some free time you may ask @gregdek to make you maintainer to make ansible even better than it is =)

**Comment by gregdek** [comment]
Thanks again to @retr0h for this PR, and thanks @grossws for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by kinghrothgar** [comment]
Has this getting merged in frozen for some reason or has it just been forgotten? I've been using a hacky work around for almost a year now that causes problems in place of this and would love to see this get merged in.

**Comment by retr0h** [comment]
@grossws any news?

**Comment by grossws** [comment]
@retr0h, I'm not a maintainer anymore but before I've stepped out of that role it was marked for inclusion.

Maybe @j2sol, @jctanner could say anything about this.

**Comment by resmo** [comment]
It seems to me this change is motivated by a special use case. I am not sure if this is the right approach. I would just expect this to be solved like:

``` yaml
- service: name={{ item }} state=restarted
  register: result
  with_items:
  - bla
  - foo
  failed_when: "result|failed and not 'Could not find the requested service' in result.msg"
```

results in 

```
TASK [service] *****************************************************************
ok: [localhost] => (item=bla) => {"failed": false, "failed_when_result": false, "item": "bla", "msg": "Could not find the requested service \"'bla'\": "}
ok: [localhost] => (item=foo) => {"failed": false, "failed_when_result": false, "item": "foo", "msg": "Could not find the requested service \"'foo'\": "}

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=0    unreachable=0    failed=0 
```

or with handler listen in a more generic way.

**Comment by omgjlk** [comment]
@resmo Scraping the output doesn't scale across all the init providers and init provider versions.

I don't see how a handler listen would solve this either.

**Comment by resmo** [comment]
`fail_when: false`?

**Comment by omgjlk** [comment]
That masks actual failures from a service that exists and fails to complete the requested action.

**Comment by gregdek** [comment]
Thanks @retr0h for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by retr0h** [comment]
@gregdek It's an entirely unmanageable process to try and get things into Ansible.

Jan 2016.

**Comment by gregdek** [comment]
@retr0h would you be willing to be a maintainer for this module? If so, I can assign to you, and then you should be able to give a shipit yourself. (Once you fix the conflict, of course.)

**Comment by retr0h** [comment]
@gregdek I'm not blaming you.  You put the `shipit` on Aug 5th.  Core is entirely way too slow, and as you see in the original PR comments.  It was accepted then reverted w/o notice.  Ansible core has now done this to me a number of times.  I'm tired of dealing with it.

**Comment by retr0h** [comment]
wut

**Comment by retr0h** [comment]
Closing, will submit a new PR.

**Pull request closed without merging**